includes/class-newspack-ads-model.php

```diff
@@ -508,5 +512,41 @@ function( $carry, $item ) {
 			[ 0, 0 ]
 		);
 	}
+
+	/**
+	 * Get ad targeting params for the current post or archive.
+	 *
+	 * @param array $ad_unit Ad unit to get targeting for.
+	 * @return array Associative array of targeting keyvals.
+	 */
+	public static function get_ad_targeting( $ad_unit ) {
+		$targeting = [];
+
+		if ( is_singular() ) {
+			// Add the post slug to targeting.
+			$slug = get_post_field( 'post_name' );
+			if ( $slug ) {
+				$targeting['slug'] = sanitize_text_field( $slug );
+			}
+
+			// Add the category slugs to targeting.
+			$categories = wp_get_post_categories( get_the_ID(), [ 'fields' => 'slugs' ] );
+			if ( ! empty( $categories ) ) {
+				$targeting['category'] = array_map( 'sanitize_text_field', $categories );
+			}
+
+			// Add the post ID to targeting.
+			$targeting['ID'] = get_the_ID();
+
+			// Add the category slugs to targeting on category archives.
+		} elseif ( 'WP_Term' === get_class( get_queried_object() ) ) {
```

Comment:
I'm getting this PHP warning when editing a post:

```
PHP Warning:  get_class() expects parameter 1 to be object, null given in /srv/www/newspack-repos/newspack-ads/includes/class-newspack-ads-model.php on line 542
```

Reply:
Update: I see the warning when loading the page on the frontend too.

Reply:
Good catch. Should be fixed in https://github.com/Automattic/newspack-ads/pull/78/commits/dced583614bac06459f07a2f5683ba51981304b1